cluster/images/provider-terraform/Dockerfile

```diff
@@ -1,3 +1,10 @@
-FROM BASEIMAGE
+FROM hashicorp/terraform:1.2.3
```

Comment:
We default to distroless now, but the original controller image used this instead of scratch.

could possibly pin to the specific digest as well.